src/psyclone/tests/domain/lfric/arg_index_to_metadata_index_test.py

```diff
@@ -0,0 +1,175 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2023, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: R. W. Ford, STFC Daresbury Lab
+
+'''This module implements tests for the ArgIndexToMetadataIndex
+class.
+
+'''
+from psyclone.domain.lfric import ArgIndexToMetadataIndex
+from psyclone.domain.lfric.kernel import (
+    ScalarArgMetadata, FieldArgMetadata, FieldVectorArgMetadata,
+    OperatorArgMetadata, ColumnwiseOperatorArgMetadata, LFRicKernelMetadata)
+
+# pylint: disable=unsubscriptable-object
```

Comment:
Please move this down to where it is required so that it doesn't affect the whole file.

Reply:
OK. Done one for each case. If I set `_info = {}` in the class as you suggest earlier it doesn't complain. However, that seems wrong seeing as the parent class sets the value.

Reply:
Yerss. Maybe this is telling us that we should be doing it differently? We could just set it directly in the subclass rather than pass it to `initialise`. In fact, given that `_initialise` is an 'internal' method, I think that actually makes sense.

Reply:
Yerss. Maybe this is telling us that we should be doing it differently? We could just set it directly in the subclass rather than pass it to `initialise`. In fact, given that `_initialise` is an 'internal' method, I think that actually makes sense.

Reply:
I think it is telling us that pylint is being a bit stupid to be honest. However, you are right that we could just set the value directly, although that does risk missing any future changes in the `_initialise` parent method. I have done that and that removes the unscriptable object complaint as pylint now knows it is a dict.